lib/dune

```diff
@@ -0,0 +1,10 @@
+(library
+  (name gen_js_api)
+  (public_name gen_js_api)
+  (synopsis "Runtime support for gen_js_api")
+  (libraries)
```

Comment:
Adding a dependencies on js_of_ocaml would probably be more accurate as gen_js_api depends on stubs provided by js_of_ocaml. @nojb, any opinion ?

Reply:
Does adding this dependency have any observable difference for users of the library, apart from requiring them to install `js_of_ocaml`?

Reply:
When compiling js from a bytecode executable, the linker will hopefully not link any of the js_of_ocaml unit as they won't be referenced anywhere (but with -linkall).

When doing separate compilation, the js_of_ocaml units will be included.

Reply:
> When doing separate compilation, the js_of_ocaml units will be included.

Will this be the case if "implicit transitive deps" in dune is false ?

Reply:
I would assume so.

Reply:
Ok, I don't feel strongly about this, but add the dependency if you think so. Otherwise we can merge directly.

Reply:
An alternative would be to generate dummy stubs similar to what jsoo does 
```
(rule
 (targets js_of_ocaml_stubs.c)
 (deps (glob_files *.ml))
 (mode promote)
 (action (with-stdout-to %{targets}
          (run ./gen_stubs/gen_stubs.exe %{deps}))))
```

Reply:
Indeed; PRs always welcome :)
Another approach would be to have `dune` pass `-no-check-prims` when an executable is compiled only in js mode.
Anyway, I think the PR is ready to be merged; improving this point can be done in a separate PR.
Shall we merge?

Reply:
Please do